### PR TITLE
Selftest after off status knobs

### PR DESCRIPTION
**Pull request by topiek-fs**
This puts the panels in "self-test" mode after the knob has been turned from off to another position.

Known issue: We can't seperate the panels yet so when the FO turns his knob to off and back to something, the captains side will also self test. This can be fixed when we can seperate the panels somehow.

### CODE REVIEW

**Comment by topiek-fs** [comment]
afbeelding
afbeelding
afbeelding

**Comment by Sxribe** [comment]
Not a maintainer, but this project uses double-quotes `"`, not single-quotes `'`. Could you update that to maintain the current used standard?

**Comment by topiek-fs** [comment]
Currently on mobile, where am I using single quotes? Can't really find it anywhere.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js [hunk 1]
```diff
@@ -31,9 +31,20 @@ class A320_Neo_EICAS extends Airliners.BaseEICAS {
...
+        this.topSelfTestLastKnobValue = 1;
+				
+        this.bottomSelfTestDiv = this.querySelector("#BottomSelfTest");
+        this.bottomSelfTestTimer = -1;
+        this.bottomSelfTestTimerStarted = false;
+        this.bottomSelfTestLastKnobValue = 1;
+		
+		this.APULastValue = false;
+		this.externalPowerLastValue = false;
+
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js:47]: Please fix indentation

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js [hunk 2]
```diff
@@ -62,6 +73,8 @@ class A320_Neo_EICAS extends Airliners.BaseEICAS {
         var engineOn = Simplane.getEngineActive(0) || Simplane.getEngineActive(1);
         var externalPower = SimVar.GetSimVarValue("EXTERNAL POWER ON", "bool");
         var apuOn = SimVar.GetSimVarValue("L:APU_GEN_ONLINE", "bool");
+		let apuStatusChanged = (apuOn != this.APULastValue);
+		let externalPowerChanged = (externalPower != this.externalPowerLastValue);
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js:77]: Please fix indentation

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js [hunk 3]
```diff
@@ -82,22 +95,52 @@ class A320_Neo_EICAS extends Airliners.BaseEICAS {
...
+            if (this.bottomSelfTestTimer <= 0) {
+                this.bottomSelfTestDiv.style.display = "none";
+				this.bottomSelfTestTimerStarted = false;
             }
         }
+		
+		this.bottomSelfTestLastKnobValue = bottomSelfTestCurrentKnobValue;
+
+		this.APULastValue = apuOn;
+		this.externalPowerLastValue = externalPower;
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js:143]: Please fix indentation

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A320_Neo_MFD.js [hunk 4]
```diff
@@ -89,9 +89,17 @@ class A320_Neo_MFD_MainPage extends NavSystemPage {
...
+
+        //SELF TEST
+        this.selfTestDiv = document.querySelector('#SelfTestDiv');
         this.selfTestTimerStarted = false;
+        this.selfTestTimer = -1;
+        this.selfTestLastKnobValueFO = 1;
+        this.selfTestLastKnobValueCAP = 1;
+
+		this.APULastValue = false;
+		this.externalPowerLastValue = false;
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A320_Neo_MFD.js:101]: Please fix indentation

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A320_Neo_MFD.js [hunk 5]
```diff
@@ -89,9 +89,17 @@ class A320_Neo_MFD_MainPage extends NavSystemPage {
         this.showILS = SimVar.GetSimVarValue("L:BTN_LS_FILTER_ACTIVE", "bool");
         this.compass.showILS(this.showILS);
         this.info.showILS(this.showILS);
-        this.selfTestDiv = this.gps.getChildById("SelfTestDiv");
-        this.selfTestTimer = -1;
+
+        //SELF TEST
+        this.selfTestDiv = document.querySelector('#SelfTestDiv');
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A320_Neo_MFD.js:94]: Here for the single quotes

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A320_Neo_MFD.js [hunk 6]
```diff
@@ -118,26 +126,44 @@ class A320_Neo_MFD_MainPage extends NavSystemPage {
...
+		let selfTestCurrentKnobValueFO = SimVar.GetSimVarValue("LIGHT POTENTIOMETER:21", "number");
+        let selfTestCurrentKnobValueCAP = SimVar.GetSimVarValue("LIGHT POTENTIOMETER:19", "number");
+		
+		if(
+            (
+                (selfTestCurrentKnobValueFO >= 0.1 && this.selfTestLastKnobValueFO < 0.1) ||
+                (selfTestCurrentKnobValueCAP >= 0.1 && this.selfTestLastKnobValueCAP < 0.1) ||
+                (externalPowerChanged || apuStatusChanged)) &&
+                !this.selfTestTimerStarted
+            ) {
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A320_Neo_MFD.js:146]: Please fix that `if` statement.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A320_Neo_MFD.js [hunk 7]
```diff
@@ -118,26 +126,44 @@ class A320_Neo_MFD_MainPage extends NavSystemPage {
...
             }
         }
+		
+		this.selfTestLastKnobValueFO = selfTestCurrentKnobValueFO
+        this.selfTestLastKnobValueCAP = selfTestCurrentKnobValueCAP
         
+        this.APULastValue = apuOn;
+        this.externalPowerLastValue = externalPower;
+
+        this.updateScreenState();
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A320_Neo_MFD.js:166]: Please fix indentation

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/PFD/A320_Neo_PFD.js [hunk 8]
```diff
@@ -131,26 +139,46 @@ class A320_Neo_PFD_MainPage extends NavSystemPage {
...
+		this.selfTestLastKnobValueFO = selfTestCurrentKnobValueFO
+        this.selfTestLastKnobValueCAP = selfTestCurrentKnobValueCAP
+
+        this.APULastValue = apuOn;
+        this.externalPowerLastValue = externalPower;
+        
+        this.updateScreenState();
     }
     onEvent(_event) {
         switch (_event) {
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/PFD/A320_Neo_PFD.js:184]: Please fix indentation and formatting issues

**Review by Benjozork — CHANGES_REQUESTED** [review]
Please configure your text editor to format your code properly.

**Comment by topiek-fs** [comment]
Resolved above

**Comment by Benjozork** [comment]
Looks way better !

Can someone test this ?

**Comment by nistei** [comment]
Test OK from cold and dark.
Test OK when loading the aircraft on the runway.

**Review by Benjozork — APPROVED** [review]

**Review by wpine215 — APPROVED** [review]

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js [hunk 9]
```diff
@@ -82,22 +95,52 @@ class A320_Neo_EICAS extends Airliners.BaseEICAS {
...
+         **/
+        
+        let topSelfTestCurrentKnobValue = SimVar.GetSimVarValue("LIGHT POTENTIOMETER:22", "number");
+        
+        if(((topSelfTestCurrentKnobValue >= 0.1 && this.topSelfTestLastKnobValue < 0.1) || (externalPowerChanged || apuStatusChanged)) && !this.topSelfTestTimerStarted) {
+            this.topSelfTestDiv.style.display = "block";
+            this.topSelfTestTimer = 14.25;
+            this.topSelfTestTimerStarted = true;
+        }
+
```
Comment (AdenFlorian) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js:109]: I think this is causing PFD and MFD to go into self test when turning on the APU, even though I'm on external power.

Also causes self test when leaving external power, even though my apu or engine is up and running.

Comment (AdenFlorian) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js:109]: https://github.com/flybywiresim/a32nx/issues/402

**Pull request merged**